src/csharp/Microsoft.Spark/Services/ConfigurationService.cs

```diff
@@ -51,6 +52,17 @@ public int GetBackendPortNumber()
             return portNumber;
         }
 
+        /// <summary>
+        /// Returns the IP address for socket communication between JVM and CLR.
+        /// </summary>
+        public string GetBackendIPAddress()
+        {
+            string ipAddress = Environment.GetEnvironmentVariable(DotnetBackendIPAddressEnvVarName);
+            _logger.LogInfo($"Using IP address {ipAddress} for connection.");
+
+            return ipAddress;
```

Comment:
Add the logic to check whether the Env variable is empty/null, and send back the appropriate address.